### ISSUE TITLE
[Panic] process::Command in tokio::test consistently causes a panic

### ISSUE THREAD

**Issue by Aobanana-chan** [tokio-rs/tokio#6934]
**Version**
tokio v1.41.0

**Platform**
Windows 10 21H2 19044.2846

**Description**
Since version 1.41, using process::Command in tokio::test consistently causes a panic. (**Debug**)

Panic on this code:
```Rust
#[tokio::test]
async fn test() {
    let out = Command::new("adb")
    .args([
        "devices"
    ])
    .output().await
    .expect("failed");

    println!("{:?}", String::from_utf8(out.stdout).unwrap());
}
```
result
![image](https://github.com/user-attachments/assets/0e70d5bc-3c57-4bb4-89b9-d948b146c028)

The message is garbled
I deciphered the garbled message, and it seems to mean "The handle is invalid."

If I replace tokio::test with tokio::main, it works well.

```Rust
#[tokio::main]
async fn main() {
    let out = Command::new("adb")
    .args([
        "devices"
    ])
    .output().await
    .expect("failed");

    println!("{:?}", String::from_utf8(out.stdout).unwrap());
}
```

I'm not sure if it's related to Tokio, but this case was working fine yesterday and only started to panic today. Interestingly, I ran the test in another crate with Tokio v1.34, and it worked perfectly. Then, I upgraded Tokio in that crate, and it started to panic just like the previous case. I tried downgrading Tokio back to 1.34, but it didn't restore the previous non-panic state.

**Comment by Aobanana-chan** [comment]
This issue did not occur on my other PC(Windows 11 23H2 22631.4317)

**Comment by Aobanana-chan** [comment]
This is so strange. It only panics when using use tokio::process::Command; in test Debug mode, but not in other cases.

https://github.com/user-attachments/assets/e2e094f2-be35-4ca3-a677-de3bce4ec037


In test Debug mode, I can use std::process::Command without issues, but I can't use tokio::process::Command. When I tried to trace the code to understand what's happening, I found this in tokio-1.41.0/src/process/windows.rs, lines 69-84:
```
pub(crate) fn spawn_child(cmd: &mut StdCommand) -> io::Result<SpawnedChild> {
    let mut child = cmd.spawn()?;
    let stdin = child.stdin.take().map(stdio).transpose()?;
    let stdout = child.stdout.take().map(stdio).transpose()?;
    let stderr = child.stderr.take().map(stdio).transpose()?;

    Ok(SpawnedChild {
        child: Child {
            child,
            waiting: None,
        },
        stdin,
        stdout,
        stderr,
    })
}
```

**Comment by Darksonn** [comment]
Have you tried with `std::process`?

**Comment by Aobanana-chan** [comment]
> Have you tried with `std::process`?

Yes, I've tried using std::process, and it works well without causing the panic that happens with tokio::process in test debug mode.

**Comment by Darksonn** [comment]
What does this do?
```rs
#[test]
fn test2() {
    std::process::Command::new("adb")
    .args(["devices"])
    .stdout(Stdio::piped())
    .stderr(Stdio::piped())
    .spawn()
    .unwrap();
}

**Comment by Aobanana-chan** [comment]
>What does this do?
> ```rust
> #[test]
> fn test2() {
>     std::process::Command::new("adb")
>     .args(["devices"])
>     .stdout(Stdio::piped())
>     .stderr(Stdio::piped())
>     .spawn()
>     .unwrap();
> }
> ```

Sorry for the late response. I tested this example and it also outputs a panic. I also tested other examples and it seems the issue lies with using the command's spawn in test debug mode. Using output works fine; only spawn causes a panic.

It seems the problem is related to std::process::Command, but this doesn't explain why this issue didn't occur before this crate. Updating Tokio to the latest version caused the problem.

**Comment by Darksonn** [comment]
If the test I asked you to run also panics, then its a bug in the standard library. You can file a bug [on the rust repository](https://github.com/rust-lang/rust) using the non-async example I gave you.

The reason I used `spawn` in the example is that this is the same as what Tokio does under the hood.

> It seems the problem is related to std::process::Command, but this doesn't explain why this issue didn't occur before this crate. Updating Tokio to the latest version caused the problem.

It probably is *not* caused by updating Tokio. You probably upgraded something else at the same time, e.g. maybe you updated the version of rustc at the same time. Based on what you told me, it's much much more likely to be due to a rustc upgrade than a Tokio upgrade.

**Comment by Aobanana-chan** [comment]
The actual situation is like this: I had two VSCode processes open simultaneously (two projects). One of them I keep open for testing purposes (like checking if other crates are usable, making simple examples, etc.), and the other is the actual project I'm working on. On the day I opened this issue, I used git pull to update the code, and then this problem occurred. I didn't upgrade rustc during this time. After roughly identifying the issue, I used another project for testing and wrote a minimal reproduction. Surprisingly, the problem didn't occur. Then, I changed the tokio version in Cargo.toml from tokio = { version = "1.34.0", features=["full"] } to tokio = { version = "1.41.0", features=["full"] }, and the panic occurred. I also didn't upgrade rustc during this period. So, I never suspected it was an issue with Rust itself.

Anyway, I will transfer this issue to the Rust repository. Thank you for your help.
